service/test/unit/resources/test_account_recovery_resource.py

```diff
@@ -67,42 +68,16 @@ def assert_error_response(_):
         d.addCallback(assert_error_response)
         return d
 
-    def test_handle_post_successfully(self):
-        request = MagicMock()
-        self.resource._get_post_form = MagicMock()
-        self.resource._validate_password = MagicMock(return_value=True)
+    def test_post_returns_failure_by_password_confirmation(self):
+        request = DummyRequest(['/account-recovery'])
+        request.method = 'POST'
+        request.content = MagicMock()
+        request.content.getvalue.return_value = '{"password": "12345678", "confirmPassword": "1234"}'
 
-        d = self.resource._handle_post(request)
+        d = self.web.get(request)
 
-        def assert_successful(success):
-            self.assertEqual(success, 'Done!')
+        def assert_error_response(_):
+            self.assertEqual(500, request.responseCode)
```

Comment:
Hey,

@thaissiqueira and me think that in case of validation error the best HTTP status is 400 (bad request), not 500.